test/gtest/ucs/test_stats.cc

```diff
@@ -253,6 +258,69 @@ class stats_on_exit_test : public stats_file_test {
     }
 };
 
+class stats_aggregate_sum_test : public stats_test {
+public:
+    virtual void init()
+    {
+        ucs_status_t status = ucs_stats_server_start(0, &m_server);
+        ASSERT_UCS_OK(status);
+        stats_test::init();
+    }
+
+    virtual void cleanup()
+    {
+        stats_test::cleanup();
+        ucs_stats_server_destroy(m_server);
+    }
+
+    virtual std::string stats_dest_config()
+    {
+        int port = ucs_stats_server_get_port(m_server);
+        EXPECT_GT(port, 0);
+        return "udp:localhost:" + ucs::to_string(port);
+    }
+
+    virtual std::string stats_trigger_config()
+    {
+        return "timer:0.1s";
+    }
+
+    void
+    read_and_check_aggrgt_sum_stats(ucs_stats_node_t *data_nodes[NUM_DATA_NODES])
+    {
+        const ucs_stats_aggrgt_sum_counter_name_t *aggregate_cnt_names_db = NULL;
+        ucs_stats_counter_t ucs_stats_aggregate_sum_counters
+                [UCS_STATS_NUM_AGGREGATE_SUM_COUNTERS_MAX];
+        size_t size;
+        size_t n_counters;
+        uint32_t i;
+
+        /* Test */
+        n_counters = ucs_stats_aggregate(ucs_stats_aggregate_sum_counters,
+                                         UCS_STATS_NUM_AGGREGATE_SUM_COUNTERS_MAX);
+
+        ucs_stats_aggregate_get_counter_names(&aggregate_cnt_names_db, &size);
+
+        ASSERT_NE(0ul, n_counters);
+        ASSERT_FALSE(aggregate_cnt_names_db == NULL);
+
+        /* Example for processing the statistics */
+        for (i = 0; i < n_counters; i++) {
+            if (aggregate_cnt_names_db[i].counter_name && aggregate_cnt_names_db[i].class_name) {
```

Comment:
why need this if?
IMO, ucs_stats_aggregate_get_counter_names already returns how many counters there are?

Reply:
Removed.